### PR TITLE
Initialize repository with collaborative codebase

### PR DESCRIPTION
**Pull request by FabianRolfMatthiasNoll**
This commit introduces a substantial codebase to the repository, which has been collaboratively developed by the entire team. This marks the inception of version control for this project using git.

### CODE REVIEW

File: src/ParametricFunction.java
Comment (kantholz44mm) [inline src/ParametricFunction.java:1]: Interfaces should be as small as possible. "Derive" only makes sense for polynomials. Getfunctionstring as well.

File: src/PolynomialFunction.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            extendCoefficientsArrayIfNeeded(exp);
+            this.coefficients[exp] = coef;
+        }
+    }
+
+    private String removeWhitespace(String str) {
+        return str == null ? null : str.trim();
+    }
+
+    private boolean isEmpty(String str) {
```
Comment (kantholz44mm) [inline src/PolynomialFunction.java]: IsEmpty does not mean "is null".

File: src/PolynomialFunction.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
+import java.util.regex.*;
+import java.util.Arrays;
+import java.util.*;
+
+public class PolynomialFunction implements ParametricFunction {
+
+    private double[] coefficients;
+    public String functionString;
+    private static final double DEFAULT_START_X = -100.0;
```
Comment (kantholz44mm) [inline src/PolynomialFunction.java]: These default values should be dynamic and represent the left and right edges of the world space (including camera position/scale)

**Review by ImFriedHof — APPROVED** [review]
Create a new Class for the Frame initialisation. This can be done by me as I will change certain aspects of it anyways.

File: src/ParametricFunction.java
Comment (FabianRolfMatthiasNoll) [reply src/ParametricFunction.java:1]: Done

File: src/PolynomialFunction.java (hunk 2 above)
Comment (FabianRolfMatthiasNoll) [reply src/PolynomialFunction.java]: Done

**Pull request merged**